backend/app/analysis/text_to_music.py

```diff
@@ -0,0 +1,179 @@
+    score = analyse_sentiment(text)
+    note_freq = 400 + (score["pos"] - score["neg"]) * 350 * (1 + score["neu"])
+
+    # get timesteps for the sample, T is note duration in seconds
+    sample_rate = 44100
+    T = 1
+    t = np.linspace(0, T, T * sample_rate, False)
+
+    # generate sine wave notes
+    note = np.sin(note_freq * t * 2 * np.pi)
+
+    # concatenate notes
+    audio = note
+    # normalize to 16-bit range
+    audio *= 32767 / np.max(np.abs(audio))
+    # convert to 16-bit data
+    audio = audio.astype(np.int16)
+
+    return wav_to_base64(audio, sample_rate)
+
+
+def get_other_freq(score, current_freq):
+    """
+    :param score: the sentiment analysis score of a sentence
+    :param current_freq: the frequency of a note
+    :return other_freq: frequency for another note found from current_freq and a randomized ratio
+    """
+    if score['neu'] > score['pos'] and score['neu'] > score['neg']:
+        ratios = neutral_ratios
+    elif score['pos'] > score['neg']:
+        ratios = consonant_ratios
+    else:
+        ratios = dissonant_ratios
+
+    ratio = random.choices(ratios)[0]
+    other_freq = current_freq * ratio[0] / (ratio[1])
+    return other_freq
+
+
+def get_notes(text):
+    """
+    :param text: a string of input text
+    :return notes: a string of notes
+    """
+    notes = ""
+    output_length = 4 if (len(text) > 30) else 2
+
+    lower_case = text.lower()
+    stripped_text = [char for char in lower_case if char in musical_chars]
+
+    if len(stripped_text) == 0:
+        notes = notes.join(random.choices(mc_list, k=output_length))
+    elif len(stripped_text) < output_length:
+        notes = notes.join(random.choices(mc_list, k=output_length-len(stripped_text))) + "".join(stripped_text)
+    else:
+        start = random.randint(0, len(stripped_text)-output_length)
+        notes = notes.join(stripped_text[start:start+output_length])
+
+    return notes
+
+
+def get_durations(notes):
+    """
+    :param notes: a list of notes
+    :return output: a list with corresponding durations
+    """
+    output = [random.uniform(0.2, 0.8) for _ in notes]
+    return output
+
+
+def sonify_sentence(text, sample_rate):
+    """
+    :param text: string of a sentence
+    :param sample_rate: integer, the sampling rate
+    :return audio: list of samples for this sentence
+    """
+    score = analyse_sentiment(text)
+    notes = get_notes(text)
+    durations = get_durations(notes)
+
+    audio = []
+
+    for index in range(len(notes)):
+        duration = durations[index]
+        note_freq = note_freqs[notes[index]]
+        other_freq = get_other_freq(score, note_freq)
+
+        time_steps = np.linspace(0, duration, int(duration * sample_rate), False)
+        louder_note = np.sin(note_freq * time_steps * 2 * np.pi).tolist()
+        quieter_note = np.sin(other_freq * time_steps * 2 * np.pi).tolist()
+
+        audio += [louder_note[ind] + 0.6 * quieter_note[ind] for ind in range(len(time_steps))]
```

Comment:
What is the `0.6` value here? I may be missing something obvious, but it might be worth pulling that out as a constant in this function to make it clear what that value is used for.

Reply:
The `0.6` value was an arbitrary constant I used to allow one note to be louder than the other, so I'll pull it out into a constant in the function.